STYLE_GUIDE.md

```diff
@@ -0,0 +1,179 @@
+# Style Guide
+
+In datarockets we enforce a community [Ruby Style Guide](https://github.com/rubocop-hq/ruby-style-guide).
+
+This is a small list of differences which we have when compared with community style guide:
+
+## Bundler
+
+* A Gem's requirements should be listed only once in a Gemfile
+
+* Gems should be alphabetically sorted within groups. Also you can use a line comment as a group separator.
+
+## Style
+
+* Limit lines to 80 characters.
+
+* Prefer double-quotes unless your string literal contains " or escape characters you want to suppress.
+
+* If elements of a hash literal span more than one line we're aligning them by keys
+
+
+```ruby
+# bad
+{
+  foo: bar,
+   ba: baz
+}
+{
+  foo: bar,
+  ba:  baz
+}
+
+# good
+{
+  foo: bar,
+  ba: baz
+}
+```
+
+* There is one exception in the rule above when the hash literal is passed as method arguments.
+In this case we're just using an indentation level.
+
+```ruby
+# bad
+do_something(foo: 1,
+             bar: 2)
+
+# good
+do_something(foo: 1,
+  bar: 2)
+```
+
+* The parameters on a multi-line method call or definition are aligning by an indentation level.
+
+```ruby
+# good
+
+def foo(bar,
+  baz)
+  123
+end
+
+def foo(
+  bar,
+  baz
+)
+  123
+end
+
+# bad
+
+def foo(bar,
+        baz)
+  123
+end
+
+# bad
+
+def foo(
+  bar,
+     baz)
+  123
+end
+```
+
+* The `end` shall be aligned with the left-hand-side of the variable assignment.
+
+```ruby
+# bad
+
+variable = if true
+           end
+
+# good
+
+variable = if true
```

Comment:
the idea is ok, but example is bad. I'd like to see something which actually works without `nil`s:
```
result = 'some cool result' if  'condition' 
```

Actually, we usually use something like this:
```
def foo(something)
  return 'bad' if something.nil?
  
  return "foo #{something}"
end
```

Reply:
I got all example from rubocop docs (See https://rubocop.readthedocs.io/en/latest/cops_layout/#layoutendalignment) 

So pls, suggest your own example if you don't like some of them

Reply:
P.S. @DDKatch 

` result = 'some cool result' if  'condition' `  is not the same 

```ruby
result = if  'condition'  
  'some cool result' 
end
```

Reply:
About the second variant, yes I like it and will add to the better section. Thoughts?

Reply:
P.S. also this is not only about `if`, we can use this rule in this example

```ruby
# good
result = array.map do |a|
  a
end

# bad

result = array.map do |a|
                 a
               end
           
```

Reply:
updated this part